lib/streamlit/user_info.py

```diff
@@ -0,0 +1,36 @@
+from streamlit.scriptrunner import get_script_run_ctx as _get_script_run_ctx
+from typing import Mapping, Dict, Optional, Iterator
+
+
+class UserInfoProxy(Mapping[str, Optional[str]]):
+    def __getitem__(self, key):
+        ctx = _get_script_run_ctx()
+        if ctx is not None:
```

Comment:
Not too important because I think that this only affects people trying to incorrectly run scripts using streamlit directly (via running `python streamlit_app.py`), but what happens here when `ctx is None`?

We may want to return a placeholder dict here similarly to how we handle the case where we don't receive user info when the websocket connects.